### PR TITLE
task/sprint 2 PIFR 26 unit test for user reducer

### PR DESCRIPTION
**Pull request by Philips7**

### CODE REVIEW

**Comment by desfero** [comment]
@Philips7 as a small git task, you need to rename those commits messages to start with `PIFR-26`

**Comment by desfero** [comment]
@Philips7 `Task` label should start from lower case

File: src/app/store/actions/user.actions.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {Action} from '@ngrx/store';
+import {User} from "../../features/users/interfaces/User";
```
Comment (desfero) [inline src/app/store/actions/user.actions.ts]: 'features'

File: src/app/store/actions/user.actions.ts [hunk 2]
```diff
@@ -17,7 +18,7 @@ export class GetUser implements Action {
 export class Authenticated implements Action {
   readonly type = AUTHENTICATED;
 
-  constructor(public payload?: any) {
+  constructor(public payload?: User) {
```
Comment (desfero) [inline src/app/store/actions/user.actions.ts]: +1

Comment (desfero) [inline src/app/store/actions/user.actions.ts]: but one question, is payload optional?

File: src/app/store/reducers/user.reduer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import {User} from '../../features/users/interfaces/User';
+import {userReducer} from './user.reducer';
+import {
+  AUTH_ERROR,
+  AUTHENTICATED, Authenticated, AuthError, GET_USER, GetUser, GoogleLogin, Logout,
+  NotAuthenticated
+} from '../actions/user.actions';
+
+describe('Testing user reducer', () => {
+
```
Comment (desfero) [inline src/app/store/reducers/user.reduer.spec.ts]: remove one line

File: src/app/store/reducers/user.reduer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import {User} from '../../features/users/interfaces/User';
+import {userReducer} from './user.reducer';
+import {
+  AUTH_ERROR,
+  AUTHENTICATED, Authenticated, AuthError, GET_USER, GetUser, GoogleLogin, Logout,
+  NotAuthenticated
+} from '../actions/user.actions';
+
+describe('Testing user reducer', () => {
```
Comment (desfero) [inline src/app/store/reducers/user.reduer.spec.ts]: simply `UserReducer`, would be quicker to indentify the place where error occur in logs

File: src/app/store/reducers/user.reduer.spec.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import {User} from '../../features/users/interfaces/User';
+import {userReducer} from './user.reducer';
+import {
+  AUTH_ERROR,
+  AUTHENTICATED, Authenticated, AuthError, GET_USER, GetUser, GoogleLogin, Logout,
```
Comment (desfero) [inline src/app/store/reducers/user.reduer.spec.ts]: sort them better, for e.g. three imports per row

File: src/app/store/reducers/user.reduer.spec.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    expect(
+      userReducer(undefined, new GetUser())
+    )
+      .toEqual({...defaultUser, loading: true});
+  });
+
+
+  it('should handle AUTHENTICATED', () => {
+    let defaultUser = new User(null, 'GUEST');
+    const action = {
```
Comment (desfero) [inline src/app/store/reducers/user.reduer.spec.ts]: probably from this object you only need payload, so can be simplified to 
````
const payload = {uid: ..., displayName: ...};

File: src/app/store/reducers/user.reduer.spec.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    )
+      .toEqual({...randomUser, loading: true});
+  });
+
+
+  it('should handle AUTH_ERROR', () => {
+    const defaultUser = new User(null, 'GUEST');
+    const action = {
+      type: AUTH_ERROR,
+      payload: 'Access Forbidden'
```
Comment (desfero) [inline src/app/store/reducers/user.reduer.spec.ts]: only payload is used, so can be simplified to `const payload = 'Access Forbidden';

File: src/app/store/actions/user.actions.ts (hunk 2 above)
Comment (Philips7) [reply src/app/store/actions/user.actions.ts]: Si, es no optional, mi error :)

File: src/app/store/reducers/user.reduer.spec.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+    expect(
+      userReducer(randomUser, new GoogleLogin())
+    )
+      .toEqual({...randomUser, loading: true});
+  });
+
+  it('should handle AUTH_ERROR', () => {
+    const defaultUser = new User(null, 'GUEST');
+    const payload =  'Access Forbidden';
```
Comment (desfero) [inline src/app/store/reducers/user.reduer.spec.ts]: one space

**Pull request merged**